cannon/mipsevm/multithreaded/state.go

```diff
@@ -255,6 +182,35 @@ func (s *State) EncodeWitness() ([]byte, common.Hash) {
 	return out, stateHashFromWitness(out)
 }
 
+func (s *State) EncodeThreadProof() []byte {
+	activeStack := s.getActiveThreadStack()
+	threadCount := len(activeStack)
+	if threadCount == 0 {
+		// There is currently one case where we will have an empty active stack with no current thread
+		// This happens at the end of wakeup thread traversal when no threads are found that are ready to wake
+		if s.Wakeup == exec.FutexEmptyAddr || !s.TraverseRight {
+			panic("Invalid empty thread stack")
+		}
+		// Just return an empty byte slice of the correct size in this case
+		return make([]byte, THREAD_WITNESS_SIZE)
+	}
+
+	activeThread := activeStack[threadCount-1]
+	otherThreads := activeStack[:threadCount-1]
```

Comment:
```suggestion
	otherThreads := activeStack[:threadCount-2]
```

Reply:
good catch!

Reply:
Wait - I think this is actually correct as-is.  The first line is grabbing the last thread in the stack. The second line is slicing the stack to remove that last thread.

Reply:
you're right. mb